pygments/formatters/html.py

```diff
@@ -510,23 +568,25 @@ def prefix(cls):
                 tmp.append((arg and arg + ' ' or '') + cls)
             return ', '.join(tmp)
 
-        styles = [(level, ttype, cls, style)
-                  for cls, (style, ttype, level) in self.class2style.items()
-                  if cls and style]
-        styles.sort()
-        lines = ['%s { %s } /* %s */' % (prefix(cls), style, repr(ttype)[6:])
-                 for (level, ttype, cls, style) in styles]
-        if arg and not self.nobackground and \
-           self.style.background_color is not None:
-            text_style = ''
-            if Text in self.ttype2class:
-                text_style = ' ' + self.class2style[self.ttype2class[Text]][0]
-            lines.insert(0, '%s { background: %s;%s }' %
-                         (prefix(''), self.style.background_color, text_style))
-        if self.style.highlight_color is not None:
-            lines.insert(0, '%s.hll { background-color: %s }' %
-                         (prefix(''), self.style.highlight_color))
-        return '\n'.join(lines)
+        return prefix
+
+    @property
+    def _pre_style(self):
+        return 'line-height: 125%; margin: 0;'
```

Comment:
Just curious - in which way `margin: 0` makes it look better? In my case, it looks less pretty because there is no margin after the code. This looks more obvious when you write some explanation on the top and code after that few times, for example:

![Oct-27-2020 18-30-12](https://user-images.githubusercontent.com/6477701/97282833-802a4100-1882-11eb-9c00-12b8785d898c.gif)

Reply:
Another example:

**With `margin: 0`**

![Screen Shot 2020-10-27 at 6 43 35 PM](https://user-images.githubusercontent.com/6477701/97284512-77d30580-1884-11eb-9109-ab9175a37bc7.png)


**Without `margin: 0`**

![Screen Shot 2020-10-27 at 6 44 02 PM](https://user-images.githubusercontent.com/6477701/97284520-7a355f80-1884-11eb-8050-8f905e8a62db.png)

I trying to override the CSS now and I was curious why this change was made.

Reply:
You're seeing (I guess) because your theme doesn't have margin/padding on the paragraph either. I'd try to override the style if you care about specific padding/margin. That said, the `margin: 0` was not intentional (I think) so I'll revert that for 2.7.3.

Reply:
Removed in 4dede4010180c8a5a231628deedcf393565cf785

Reply:
Thanks @Anteru :).

Reply:
Hi! I'm sorry if I'm a bit too late, but I'm pretty I've introduced the `margin: 0` rule for a reason. Since I've worked on this one some time ago, I'm not sure what exactly was it, but I'd guess it had something to do with making the table and non-table styles look more similar. An alternative might be that it had something to do with highlighting lines/special lines. There was a lot of different combinations of styles and highlight rules etc. and I believe `margin: 0` was introduced to make everything work better in regard to some of these combinations.

Having said that, if it actually looks better without `margin: 0` and it doesn't break anything else, I'm totally OK with removing this rule :)

Reply:
I can't find the case where it would matter, and it does break various themes -- if you can point me to a problematic case then we can revisit alternative solutions. Ideally theme authors should have a way to override Pygments CSS rules, but for now this seems to be causing too much fallout :(

Reply:
I tried to add all problematic cases to the test suite :P But while working on this a lot of dependencies came up and it's possible I've changed one thing too many ;) If you could just add a proper test with an explanation/justification, I would be very grateful. Right now, unfortunately, I'm not able to spend any time fixing this or checking why exactly was this change made. Reverting it seems like a good solution, as long as it's documented ;)

Reply:
Fair enough. I ran a bunch of manual tests and also spot-checked the test cases, and couldn't find a problematic case. Was hoping you'd remember something particularly problematic, but given the feedback we got, it seems that any rule changing the margin will cause more harm than good :(